### PR TITLE
HBASE-26518 Concurrent assign and SCP may cause regionserver …

### PR DESCRIPTION
**Pull request by sunhelly**
…abort

More details in the description of the issue https://issues.apache.org/jira/browse/HBASE-26518

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 14s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 47s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 47s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 41s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 19s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  8s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  5s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 43s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 51s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 51s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 31s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  0s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m  9s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   3m  2s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 131m 38s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 179m 58s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3900 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux dca6ac00d4bc 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3900/out/precommit/personality/provided.sh |
| git revision | branch-1 / 13787ae03c |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/testReport/ |
| Max. process+thread count | 4680 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 26s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 48s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 51s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 57s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 32s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 42s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 16s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 50s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 22s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  38m 19s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   3m 14s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 155m  9s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 234m 53s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.mapreduce.TestLoadIncrementalHFiles |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3900 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 595c2f6280dc 4.15.0-162-generic #170-Ubuntu SMP Mon Oct 18 11:38:05 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-3900/out/precommit/personality/provided.sh |
| git revision | branch-1 / a2406f3764 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/testReport/ |
| Max. process+thread count | 4104 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3900/1/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStates.java [hunk 1]
```diff
@@ -824,6 +824,11 @@ public void regionOffline(
     }
 
     for (HRegionInfo hri : regionsToOffline) {
```
Comment (GeorryHuang) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStates.java:826]: I saw your description in https://issues.apache.org/jira/browse/HBASE-26518, it says this issue could be reproduced. So we should also write a UT about it?

**Comment by Apache9** [comment]
The same with HBASE-25783, we plan to EOL branch-1 so I tend to not include the changes which touch the core part.

**Pull request closed without merging**